purchases/src/defaults/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -839,6 +839,29 @@ class Purchases internal constructor(
             purchasesOrchestrator.allowSharingPlayStoreAccount = value
         }
 
+    /**
+     * The preferred UI locale override for RevenueCat UI components.
+     * This affects both API requests and UI rendering.
+     *
+     * @return The preferred UI locale override, or null if using system default
+     */
+    val preferredUILocaleOverride: String?
+        @Synchronized get() = purchasesOrchestrator.preferredUILocaleOverride
+
+    /**
+     * Override the preferred UI locale for RevenueCat UI components at runtime.
+     * This affects both API requests and UI rendering.
+     *
+     * If the locale changes, this will automatically clear the offerings cache and trigger
+     * a background refetch to get paywall templates with the correct localizations.
+     *
+     * @param localeString The locale string (e.g., "es-ES", "en-US") or null to use system default
+     * @return true if cache was cleared and refetch triggered, false if locale unchanged or cache clear rate limited
+     */
+    fun overridePreferredUILocale(localeString: String?): Boolean {
```

Comment:
Hmm is there a need for this method to return a boolean? Feels like this should not return anything IMO. It would also simplify the code a bit. Wdyt?